src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -812,8 +847,185 @@ def __set_errors_json(self, error_count_by_operation, errors_by_operation):
+            status_file_payload_json_dumps = json.dumps(truncated_status_file)
+
+        return status_file_payload_json_dumps
+
+    def __create_truncated_status_file(self, status_file_size_in_bytes, complete_status_file_payload_json):
+        """ Truncate substatus message patch list when complete status file size is more than 126kb """
+        """
+            __create_truncated_status_file(self, status_file_size_in_bytes, complete_status_file_payload_json):
+
+                truncated_status_file = json.loads(complete_status_file_payload_json)
+                low_pri_index = None
+                _index = self.__get_substatus_index()
+                status_file_without_package_list_size  = __calc_package_payload_size_on_disk(size_of_constant_status_data(complete_status_file_payload_json))
+                size_of_max_packages_allowed_in_status = 126kb - status_file_without_package_list_size 
+                
+                if assessment_index is not none:
+                    _substatus_msg_copy = __get_substatus_message(truncated_status_file, _index)
+                    _packages_copy = _substatus_msg_copy['patches']
+
+                if installation_index is not none:
+                    _substatus_msg_copy = __get_substatus_message(truncated_status_file, _index)
+                    _packages_copy = _substatus_msg_copy['patches']
+                    low_pri_index = __get_installation_low_pri_index()
+
+                    while status_file_size_in_bytes > 126kb:
+                        __apply_truncation_process()
+                            __split_assessment_list()
+                                __apply_truncation()
+                                
+                        __recompose_truncated_status_file()                                    
+                            __get_errors_from_substatus()
+                            __recompose_truncated_substatus_msg()
+                                __recompose_substatus_msg_errors()
+                                __create_assessment_tombstone_list()
+                                    __create_assessment_tombstone()
+                                __recreate_assessment_summary_json()
+                       
+                        __recompose_truncated_status_file()                                    
+                            __get_errors_from_substatus()
+                            __recompose_truncated_substatus_msg()
+                                __recompose_substatus_msg_errors()
+                                __create_installation_tombstone
+                                __recreate_installation_summary_json()
+                                
+                        status_file_size_in_bytes, status_file_agent_size_diff = __get_new_size_in_bytes_after_truncation(truncated_status_file)
+                        size_of_max_packages_allowed_in_status -= status_file_agent_size_diff  
```

Comment:
Reference from original PR, make sure to remove the algorithm before merging:

<img width="887" alt="image" src="https://github.com/Azure/LinuxPatchExtension/assets/56841542/fcf4d1d0-85d7-40d4-ac10-39ef291e71c4">